test/file_io.jl

```diff
@@ -148,14 +148,21 @@ end
 end
 
 @testset "saving and loading" begin
-    mktempdir() do path
+    function test_save_and_load(dirpath)
         @testset "$fmt - $k" for fmt in (:bson, :julia_serialize), (k, v) in datas
-            JLSO.save("$path/$fmt-$k.jlso", k => v; format=fmt)
-            result = JLSO.load("$path/$fmt-$k.jlso")
-            single_item = JLSO.load("$path/$fmt-$k.jlso", k)
+            filepath = joinpath(dirpath, "$fmt-$k.jlso")
+            JLSO.save(filepath, k => v; format=fmt)
+            result = JLSO.load(filepath)
+            single_item = JLSO.load(filepath, k)
             @test result[k] == single_item[k] == v
         end
     end
+    @testset "String-type paths" begin
+        test_save_and_load(mktempdir())
```

Comment:
should these be the other way round (does it matter?)
```suggestion
        mktempdir(test_save_and_load())
```

Reply:
I think so? because then it clens up immediately?